### PR TITLE
Respect trailing newline at the end of the file

### LINKED ISSUE #26: New line at EOF
I'm noticing that this adds a new line at the end of the file even when the existing JSON file doesn't have one. I am using this in `detectIndent` so I expected the whitespace to be left untouched.

Perhaps it could respect the EOF by default, or add a `detectEOF` option?

Happy to open a PR if we can decide on a direction with this issue.

### PR DESCRIPTION
**Pull request by privatenumber**
Fixes #26

This PR adds functionality to detect EOF and respect it

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,6 +10,11 @@ const isPlainObj = require('is-plain-obj');
 
 const readFile = promisify(fs.readFile);
 
+const detectEOF = file => {
```
Comment (sindresorhus) [inline index.js]:
```suggestion
const hasTrailingNewline = file => {
```

File: index.js [hunk 2]
```diff
@@ -10,6 +10,11 @@ const isPlainObj = require('is-plain-obj');
...
+	const detected = file.match(/\r?\n$/);
+	return detected ? detected[0] : '';
```
Comment (sindresorhus) [inline index.js]: Doesn't really make sense to use the result, as we won't preserve `\r\n` for the rest of the JSON anyway, so you can just use `\n`.

File: index.js [hunk 3]
```diff
@@ -37,40 +42,42 @@ const init = (fn, filePath, data, options) => {
 
 const main = async (filePath, data, options) => {
 	let {indent} = options;
-
-	if (options.detectIndent) {
-		try {
-			const file = await readFile(filePath, 'utf8');
+	let EOF = '\n';
```
Comment (sindresorhus) [inline index.js]: `EOF` is not the correct term for this.

File: index.js (hunk 2 above)
Comment (privatenumber) [reply index.js]: Good point

File: index.js (hunk 3 above)
Comment (privatenumber) [reply index.js]: Why not? End-of-file? Doesn't it make sense in context?
```
`${json}${EOF}`
```

And if not, what do you prefer? `trailingNewline`?

Comment (sindresorhus) [reply index.js]: EOF is a special character: https://en.wikipedia.org/wiki/End-of-file

> trailingNewline?

Yes

**Comment by sindresorhus** [comment]
The PR needs a better title too.

**Comment by privatenumber** [comment]
@sindresorhus Addressed all review comments

**Pull request merged**